### DISCUSSION TITLE
Kotlin: Lateinit property engine not initialized

### DISCUSSION THREAD

**Discussion by reflqct** [AlmasB/FXGL#1124] [Q&A]
Hi!

I'm coding a game, using java on a mac. When I added this line: ```
FXGL.getGameWorld().addEntityFactory(new LevelLoader());
FXGL.setLevelFromMap("Ruby - 111aaa.tmx");
```
My code got an error: ```Caused by: kotlin.UninitializedPropertyAccessException: lateinit property engine has not been initialized```

How can I fix this?

**Comment by AlmasB** [comment]
Hi @RubyCodesYT 
Typically means FXGL.* calls are being made before the engine is ready. For example, in `initGame()` it is safe to call FXGL, but not in:

```
class A extends GameApplication {

// here, at top-level, not safe as engine is not ready.

initGame() { // safe }
}
```

Hope this helps. If not, feel free to post your code around the calls above.
